torch/distributed/fsdp/flat_param.py

```diff
@@ -1499,7 +1509,7 @@ def prepare_gradient_for_optim(self):
         def cast_grad_to_param_dtype_if_needed(flat_param):
             # TODO (rohan-varma): test for full precision with keep_low_precision_grads
             if not self._force_full_precision and self._keep_low_precision_grads:
-                assert flat_param.grad is not None  # mypy
+                _p_assert(flat_param.grad is not None, "Unexpected None grad!")
```

Comment:
It should not, but might be useful to keep to avoid introducing bugs in the future. I hit this assert when testing, and it was hard to isolate because we were not using `p_assert`